optax/_src/transform.py

```diff
@@ -351,7 +353,16 @@ def update_fn(updates, state, params=None):
     mu = update_moment(updates, state.mu, b1, 1)
     nu = update_moment_per_elem_norm(updates, state.nu, b2, 2)
     count_inc = numerics.safe_int32_increment(state.count)
-    mu_hat = bias_correction(mu, b1, count_inc)
+    if nesterov:
+      mu_hat = jax.tree_util.tree_map(
+          lambda m, g: b1 * m + (1 - b1) * g,
+          bias_correction(mu, b1, numerics.safe_int32_increment(count_inc)),
+          bias_correction(updates, b1, count_inc))
+      # Dozat 2016 https://openreview.net/pdf?id=OM0jvwB8jIp57ZJjtNEZ
+      # Algorithm 2 further multiplies Adam's standard nu_hat by b2. It is
+      # unclear why. Other Nadam implementations also omit the extra 2 factor.
```

Comment:
```suggestion
      # unclear why. Other Nadam implementations also omit the extra b2 factor.
```

Reply:
ooops, I thought that was a typo, but obviously I had misread

Reply:
Yep it isn't. For future reference, it corresponds to this $\nu$
![image](https://github.com/google-deepmind/optax/assets/13407481/860b1267-9bf9-4368-aaba-790d64bf3f9b)